code/controllers/subsystem/mapping.dm

```diff
@@ -257,10 +257,13 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 /datum/controller/subsystem/mapping/proc/generate_station_area_list()
 	var/list/station_areas_blacklist = typecacheof(list(/area/space, /area/mine, /area/ruin, /area/asteroid/nearstation))
 	for(var/area/A in world)
-		var/turf/picked = safepick(get_area_turfs(A.type))
-		if(picked && is_station_level(picked.z))
-			if(!(A.type in GLOB.the_station_areas) && !is_type_in_typecache(A, station_areas_blacklist))
-				GLOB.the_station_areas.Add(A.type)
+		if (is_type_in_typecache(A, station_areas_blacklist))
+			continue
+		if (!A.contents.len)
+			continue
+		var/turf/picked = A.contents[1]
+		if (is_station_level(picked.z))
+			GLOB.the_station_areas |= A.type
```

Comment:
+= should be fine unless areas share types for some reason. And even then, my understanding is GLOB.the_station_areas[A.type] = 0 would be more efficient anyway.

Reply:
Areas do share types, in particular escape pods use multiple instances of the same area. Will double-check WRT associative list for this though.